torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -91,6 +92,74 @@ void checkSameSizeAndType(
   }
 }
 
+void checkSplitSizes(
+    const std::vector<int64_t>& split_sizes,
+    const at::Tensor& tensor,
+    int group_size) {
+  if (split_sizes.size() == 0) {
+    if (tensor.size(0) % group_size != 0) {
+      throw std::runtime_error(
+          "Tensor's dim 0 does not divide equally across group size");
+    }
+  } else {
+    if (split_sizes.size() != group_size) {
+      throw std::runtime_error("Number of tensor splits not equal to group size");
```

Comment:
Same here and below, let's replace all throw into `TORCH_CHECK` or `TORCH_INTERNAL_ASSERT`